applications/HDF5Application/python_scripts/core/operations/model_part.py

```diff
@@ -299,104 +325,132 @@ class PrimalBossakOutput(VariableIO):
 
-class PrimalBossakInput(VariableIO):
+
+class PrimalBossakInput(VariableIOOperation):
     '''Reads nodal solution step data from a file.
 
     This is used by the transient adjoint solvers.
     '''
 
-    def __init__(self, settings):
-        super(PrimalBossakInput, self).__init__(settings)
-
-    def __call__(self, model_part, hdf5_file):
-        primal_io = KratosHDF5.HDF5NodalSolutionStepBossakIO(
-            self.GetSettings(model_part).Get(), hdf5_file)
-        primal_io.ReadNodalResults(
-            model_part)
+    def Execute(self) -> None:
+        KratosHDF5.HDF5NodalSolutionStepBossakIO(
+            self.parameters,
+            self.file).ReadNodalResults(self.model_part)
 
 
-class MoveMesh:
+class MoveMesh(ModelPartIOOperation):
     '''Perform a mesh move operation on a model part.
 
     The primary use case is to set the mesh to the current configuration after
     reading the model part.
     '''
 
-    def __call__(self, model_part, *args):
-        KratosMultiphysics.ImplicitSolvingStrategy(model_part, True).MoveMesh()
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        print("HDF5 MoveMesh operation is deprecated and will be removed", file = sys.stderr)
 
+    def Execute(self) -> None:
+        KratosMultiphysics.ImplicitSolvingStrategy(self.model_part, True).MoveMesh()
 
-def Create(settings):
-    '''Return the operation specified by the setting 'operation_type'.
+
+def GetSubclasses(base_class: type) -> "list[type]":
+    """Recursively find all subclasses of a base class"""
+    subclasses = base_class.__subclasses__()
+    for subclass in base_class.__subclasses__():
+        subclasses += GetSubclasses(subclass)
+    return subclasses
+
+
+class AggregateOperation(KratosMultiphysics.Operation):
+    """ @brief Class for aggregating HDF5 IO operations on the same file."""
+
+    def __init__(self, model: KratosMultiphysics.Model, parameters: typing.Union[KratosMultiphysics.Parameters, ParametersWrapper]):
+        super().__init__()
+        if isinstance(parameters, ParametersWrapper):
+            parameters = parameters.Get()
+        parameters.AddMissingParameters(self.GetDefaultParameters())
+
+        self.__model_part = model.GetModelPart(parameters["model_part_name"].GetString())
+        self.__io_parameters = parameters["io_settings"]
+
+        # {operation_type, operation_parameters}
+        self.__operations: "list[tuple[type, KratosMultiphysics.Parameters]]" = []
+        for i in range(parameters["list_of_operations"].size()):
+            self.Add(parameters["list_of_operations"][i])
+
+    def Execute(self) -> None:
+        with OpenHDF5File(self.__io_parameters, self.__model_part) as file:
+            for operation, operation_parameters in self.__operations:
+                operation(self.__model_part, operation_parameters, file).Execute()
+
+    def Add(self, operation_parameters: KratosMultiphysics.Parameters) -> None:
```

Comment:
Is there a reason why this `Add` is public? Are we gonna allow it? Does it make sense?